Update SuperTuxKart to 0.9.3

ahayzen opened the pull request:
Please note this is my first patch to a flatpak project, so I've probably done some things wrong :-)

I had issues with Fribidi building on my machine.

Firstly the previous location of it (http://fribidi.org/download/fribidi-0.19.7.tar.bz2) states "Not Found", so I updated it to the github url.

Then it failed to compile due to multiple reasons:
- "-g" being an unrecognised option for configure (I have removed from cflags)
- .Tpo's missing (there were .Po instead) `mv: cannot stat '.deps/packtab.Tpo': No such file or directory` (---disable-dependency-tracking resolved this)
- c2man being missing, this is related to the docs and seems simplest to skip as stated here https://github.com/fribidi/fribidi/issues/8 so I copied the gentoo patch to disable building the docs https://sources.gentoo.org/cgi-bin/viewvc.cgi/gentoo-x86/dev-libs/fribidi/files/fribidi-0.19.2-nodoc.patch?revision=1.2&view=markup
- Some compiler warnings and deprecated declarations (modified the cflags in the configure)

Then after updating SuperTuxKart from 0.9.2 to 0.9.3 the following message was shown
```
CMake Error at CMakeLists.txt:163 (message):
  libopenglrecorder not found.  Either install libopenglrecorder or disable
  in-game recorder with -DBUILD_RECORDER=0
```

To get it building i've disabled the in-game recorder, but happy to add the lib if required?

Below is a screenshot of SuperTuxKart 0.9.3 running :-)

![screenshot from 2017-11-26 21-15-21](https://user-images.githubusercontent.com/3669750/33244557-d22bcf16-d2f1-11e7-927f-ca22a1d27fcc.png)

ahayzen commented:
Also I saw that Fribidi is used in some other flatpaks (https://github.com/search?q=org%3Aflathub+fribidi&type=Code) I wonder if those will also need the download path updating? And potentially some of the patches applied?

TingPing commented:
bot, build

TingPing commented:
> Also I saw that Fribidi is used in some other flatpaks (https://github.com/search?q=org%3Aflathub+fribidi&type=Code) I wonder if those will also need the download path updating? And potentially some of the patches applied?

Might be worth adding to flathub/shared-modules

TingPing commented on fribidi-ignore-warnings.patch:
```diff
@@ -0,0 +1,12 @@
+ changequote(,)dnl
+ if test "x$GCC" = "xyes"; then
+-  CFLAGS="$CFLAGS -Wall -ansi "
++  CFLAGS="$CFLAGS -ansi -Wno-deprecated-declarations"
```
The better solution to this is finding the `-Werror` and removing it.

ahayzen replied:
With `-Werror` it fails like this https://pastebin.ubuntu.com/26053660/

It is therefore acceptable to have the CFLAGS as `CFLAGS="$CFLAGS -Werror -Wno-deprecated-declarations -ansi "` ? (As this then works)

Barthalion replied:
In other words, the problem is that there is -Werror defined somewhere and it shouldn't be except for development purposes. It should be removed instead of adding warious `-Wno-` flags.

ahayzen replied:
Ah! I think the `-Wno-deprecated-declarations` isn't actually required, I just had it there when trying to use `-Werror` as well. OK I'll remove that `-Wno-deprecated-declarations` so the cflags can become `CFLAGS="$CFLAGS -ansi "`. (I hope this is what you mean, and I haven't got confused).

ahayzen commented:
I'm going to see how simple libopenglrecorder is to get included as well, as this then allows for in-game recording I believe?

ahayzen commented:
OK, so I found that the warnings patch wasn't actually required (the docs patch probably fixed what I was trying to fix :-) ), so I have removed that.

I have added libopenglrecorder to the build and enabled recording, following the instructions here https://supertuxkart.net/Artist_Debug_Mode#In-game_Screen-recorder I was able to spit out a webm of gameplay.

Now I just have a weird situation on my machine, that I hope is a caching issue or something as I've been updating/rebuilding the app many times. After installing the app from my local repository, if I run the app the "normal" way I get the following:
```
$ flatpak run net.supertuxkart.SuperTuxKart 
supertuxkart: error while loading shared libraries: libturbojpeg.so.0: cannot open shared object file: No such file or directory
```

But if I run from the shell it works fine
```
$ flatpak run --devel --command=sh net.supertuxkart.SuperTuxKart 
sh-4.3$ supertuxkart 
[verbose  ] main: Error messages and other text output will be logged to /home/andrew/.var/app/net.supertuxkart.SuperTuxKart/config/supertuxkart/0.8.2/stdout.log.
```
And I can see `libturbojpeg` in `/lib`

ahayzen commented:
Ah without the `--devel` it also fails using the command, does this mean that the --devel is bringing in the SDK and then libturbojpeg ? So then I need to include libturbojpeg myself ?

ahayzen commented:
Resolved the issue, I have added libjpeg-turbo to the modules.

Please re-review now, it should be in a good state :-)

alexlarsson commented:
Hmm, it seems that the sdk contains libjpeg62 only, but the sdk contains libjpeg-dev, libjpeg62 *and* libturbojpeg0. Note that our libjpeg62 is still libjpeg-turbo, but without the turbo ABI.

I guess we should add the turbo one, as we're shipping the headers for it.

knurd commented:
@ahayzen FYI (FWIW): I also looked into updating to 0.9.3 before seeing this. I noticed you can avoid all the hassle with fribidi if you stick to the older tarball (pre-github move), which I for example found at https://ftp.osuosl.org/pub/blfs/conglomeration/fribidi/fribidi-0.19.7.tar.bz2

ahayzen commented:
@knurd thanks for the info, but I guess as we've managed to get it working with the 'new' github releases that'll be the preferred route.

ahayzen commented:
`libjpeg-turbo` is now in the runtime, so I have removed this from the modules and have checked that the app launches correctly.

TingPing commented:
bot, build
